hivemind/utils/asyncio.py

```diff
@@ -167,12 +169,24 @@ async def attach_event_on_finished(iterable: AsyncIterable[T], event: asyncio.Ev
 class _AsyncContextWrapper(AbstractAsyncContextManager):
     """Wrapper for a non-async context manager that allows entering and exiting it in EventLoop-friendly manner"""
 
+    EXECUTOR_PID = None
+    CONTEXT_EXECUTOR = None
+    EXECUTOR_LOCK = mp.Lock()
+
     def __init__(self, context: AbstractContextManager):
         self._context = context
 
+    @classmethod
+    def get_process_wide_executor(cls):
+        if os.getpid() != cls.EXECUTOR_PID:
+            with cls.EXECUTOR_LOCK:
```

Comment:
```suggestion
        with cls.EXECUTOR_LOCK:
            if os.getpid() != cls.EXECUTOR_PID:
```

Reply:
Yeah, I agree that the double if is better in this case :)

Reply:
implemented the latter option with double checked locking